sdk/src/main/java/tech/relaycorp/sdk/StorageImpl.kt

```diff
@@ -0,0 +1,85 @@
+package tech.relaycorp.sdk
+
+import org.bouncycastle.jce.provider.BouncyCastleProvider
+import tech.relaycorp.relaynet.wrappers.x509.Certificate
+import tech.relaycorp.sdk.persistence.Persistence
+import java.security.KeyFactory
+import java.security.KeyPair
+import java.security.PrivateKey
+import java.security.interfaces.RSAPrivateCrtKey
+import java.security.spec.PKCS8EncodedKeySpec
+import java.security.spec.RSAPublicKeySpec
+import javax.inject.Inject
+
+internal class StorageImpl
+@Inject constructor(
+    private val persistence: Persistence
+) {
+
+    private suspend fun getPrivateKey(endpoint: String) =
+        persistence.get("private_key_$endpoint")?.toPrivateKey()
+
+    private suspend fun setPrivateKey(endpoint: String, privateKey: PrivateKey) {
+        persistence.set("private_key_$endpoint", privateKey.encoded)
+    }
+
+    private suspend fun deletePrivateKey(endpoint: String) {
+        persistence.delete("private_key_$endpoint")
+    }
+
+    suspend fun getKeyPair(endpoint: String) =
+        getPrivateKey(endpoint)?.toKeyPair()
+
+    suspend fun setKeyPair(endpoint: String, keyPair: KeyPair) {
+        setPrivateKey(endpoint, keyPair.private)
+    }
+
+    suspend fun deleteKeyPair(endpoint: String) {
+        deletePrivateKey(endpoint)
+    }
+
+    suspend fun getCertificate(endpoint: String) =
+        persistence.get("certificate_$endpoint")?.let(Certificate::deserialize)
```

Comment:
To future-proof things, I'd suggest to further qualify "key pair" and "certificate" as "identity key pair" and "identity certificate". Otherwise, these things will become ambiguous when we implement the [channel session protocol](https://specs.relaynet.network/RS-003), which also requires key pairs and certificates.